studio/components/layouts/ProjectLayout/NavigationBar/NavigationBar.tsx

```diff
@@ -21,6 +21,28 @@ const NavigationBar: FC<Props> = ({}) => {
   const productRoutes = generateProductRoutes(projectRef)
   const otherRoutes = generateOtherRoutes(projectRef)
 
+  const [value, setValue] = useState('')
```

Comment:
@harshcut can we abstract these pieces of logic for the theme state, useEffect, and onThemeOptionChange into the UiStore?

The NavigationBar and  /account/me have the same pieces of logic so ideally we'd not want to repeat the code in both places. i.e the UiStore should be the source of truth, it should hold the `themeOption`, and NavigationBar and /account/me should just listen to that value instead of holding its own state

`onThemeOptionChange` can be placed within the UiStore as well and the `useEffect` can probably be placed in the `load` method of the UiStore